### PR TITLE
qa: Explain QT_QPA_PLATFORM for gui tests

### LINKED ISSUE #17013: macOs: "test_bitcoin-qt -platform cocoa" no longer works
As of v0.18.1 I can use `test_bitcoin-qt -platform cocoa` to run GUI tests; this opens and closes a regtest GUI a bunch of times.

As of master (6e4f6553913978fcdddba30e0bb6f97edcb12e80) it seems to ignore the `-platform cocoa` and just run the test as with `minimal`.

As seen on macOS 10.14.6. Git bisect blames a2714a5c69f0b0506689af04c3e785f71ee0915d in #16578 `Do not pass in command line arguments to QApplication` (@achow101, @ryanofsky)

### PR DESCRIPTION
**Pull request by maflcko**
Closes #17013

### CODE REVIEW

**Review by Sjors — APPROVED** [review]
Code review ACK faccf5f, and tested QT_QPA_PLATFORM=cocoa on macOS.

File: src/qt/test/test_main.cpp [hunk 1]
```diff
@@ -63,7 +63,7 @@ int main(int argc, char *argv[])
     #if defined(WIN32)
         _putenv_s("QT_QPA_PLATFORM", "minimal");
```
Comment (ryanofsky) [inline src/qt/test/test_main.cpp]: Would suggest adding `if (getenv("QT_QPA_PLATFORM") == nullptr)` so the new instructions for setting `QT_QPA_PLATFORM` work on windows. Or, if you'd prefer to not change behavior in a documentation-only PR, you could add a `// TODO: Fix this to not overwrite QT_QPA_PLATFORM if it is set externally` comment

**Review by ryanofsky — APPROVED** [review]
utACK faccf5f9c899c40d4da5792629d0714249a4616b

File: src/qt/test/test_main.cpp (hunk 1 above)
Comment (MarcoFalke) [reply src/qt/test/test_main.cpp]: Thx, added a commit and changed the pull request type to "qa"

**Review by promag — COMMENTED** [review]
ACK fa06bb607da2d3e35723661156d873c8eac1fa50.

**Review by jonasschnelli — APPROVED** [review]
ACK fa06bb60

**Review by ryanofsky — APPROVED** [review]
utACK fa06bb607da2d3e35723661156d873c8eac1fa50

**Review by fanquake — APPROVED** [review]
ACK fa06bb607da2d3e35723661156d873c8eac1fa50 - tested on macOS using `QT_QPA_PLATFORM=cocoa src/qt/test/test_bitcoin-qt`.

**Pull request merged**